lib/src/main/scala/spinal/lib/bus/bmb/BmbDecoder.scala

```diff
@@ -51,7 +51,11 @@ case class BmbDecoder(p : BmbParameter,
     input.ready := (hitsS1, io.outputs).zipped.map(_ && _.cmd.ready).orR || noHitS1
 
     val rspPendingCounter = Reg(UInt(log2Up(pendingMax + 1) bits)) init(0)
-    rspPendingCounter := rspPendingCounter + U(input.lastFire) - U(io.input.rsp.lastFire)
+    if (p.access.canSync)
+      rspPendingCounter := (rspPendingCounter + U(input.lastFire) - U(io.input.rsp.lastFire)
+                           + U(input.lastFire && input.isWrite) - U(io.input.sync.valid && input.isWrite))
```

Comment:
I think - U(io.input.sync.valid && input.isWrite) is wrong. input.isWrite isn't related.
- U(io.input.sync.valid)    should be enough, as sync only come for writes already.

Reply:
Also, io.input.sync.valid isn't right, instead, do io.input.sync.fire to ensure it is high only once cycle